components/layout_2020/dom_traversal.rs

```diff
@@ -282,7 +285,20 @@ where
             for item in items.iter() {
                 match item {
                     ContentItem::String(s) => {
-                        vec.push(PseudoElementContentItem::Text(s.to_string()))
+                        vec.push(PseudoElementContentItem::Text(s.to_string()));
+                    },
+                    ContentItem::Attr(attr) => {
+                        let element = element
+                            .to_threadsafe()
+                            .as_element()
+                            .expect("Expected an element");
+                        let attr_val = element.get_attr(
+                            &attr.namespace_url,
+                            &LocalName::from(attr.attribute.to_string()),
```

Comment:
```suggestion
                            &LocalName::from(&*attr.attribute),
```

`style::Atom` dereferences to `&str` and `LocalName` implemements `From<&str>`, so there is no need to allocate a string here.